### ISSUE TITLE
Remove the "Build project" entry from the project explorer context menu

### ISSUE THREAD

**Issue by pounard** [eclipse/pdt#108]
"Build project" option from Eclipse doesn't really make sense for most PHP projects. And when we trigger it, the operation is blocking and prevent using the IDE until it finishes.

Problem is that, on large projects, even with a more than decent and more than recent machine (i.e. a nice core i7 with 16 go of RAM, and a beefed-up Eclipse configuration for using much RAM) this operation can take up to 20~30 minutes, using all CPU cores and making the machine lag (it happened to me right now).

Aside the fact that it forces me to do an intellectual break from my code (which is not a bad thing) and the fact that it's so long I decided to take some time to write this issue, it's quite annoying mostly.

I have no idea why it's so slow, but in the many tasks it does, between a full PHP code re-indexing (it seems at least) and validating all HTML files it finds (and sadly, last time I accidentally triggered it, I had some coverage HTML report files, a huge HTML report).

Fact is, it actually happens a lot, accidentally triggering this useless feature, either by misclicking it in the contextual menu, or by accidentally typing the keyboard shortcut for it (yes, it happens to me a lot).

When Eclipse doesn't crash on the most extreme projects I work with, I have to wait for 20~30 minutes before being able reusing my IDE.

Is there a way to either:

 - completely disable this feature for PHP projects,
 - or fine tune it to avoid that terrible itch it causes on a regular basis.

Thanks. in advance.

**Comment by zulus** [comment]
In general this process do multiple things:

1. Provoke indexing
2. Syntax validation (PHP, CSS, JSON etc...)
3. Semantic validation (variables, class/trait/inteface) etc..
4. Task list like @TODO

Problem is because after crash, eclipse by default invalidate everything especially index.

Build is also required after outside project change (like composer install) due DLTK bug. Once DLTK will be completely removed will be more stable

**Comment by mlocati** [comment]
Isn't it the same as `Project` &rarr; `Clean...`?

![immagine](https://user-images.githubusercontent.com/928116/119671439-0bc87800-be3a-11eb-879c-c09a0e59a3f2.png)

**Comment by pounard** [comment]
My main issue with it is that in most cases, I trigger it accidentally. If the process was less consuming it wouldn't be a problem. Or if it was harder to trigger it accidentally it would be a nice thing as well.

**Comment by mlocati** [comment]
What about changing the keyboard shortcut from
`Ctrl`+`B`
to
`Left Shift`+`Right Shift`+`Ctrl`+`Alt`+`AltGr`+`Sing a specific song`+`B`
? 🤣

**Comment by pounard** [comment]
> Normally this button provoke "incremental build" (file changes since last build only), but when you disable auto-build it's mostly "full build" (all files).

Rebuild doesn't clear all and do a full rebuild ?

**Comment by zulus** [comment]
> Rebuild doesn't clear all and do a full rebuild ?

Yeah, but most of builders have 3 different scenarios for clean/full/incremental builds.

Anyway problem is different. Any build operation locks DLTK and WTP, editor (and entire eclipse) is mostly useless.
